### PR TITLE
feat: Validation input comps

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/components/inputs/text/ArweavetransactionIdInput/ArweavetransactionIdInput.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  placeholder?: string;
+}) {
+  useEffect(() => {
+    setValidation([
+      // set default validations on load
+      {
+        'Is a transaction ID': {
+          result: undefined,
+          error: '',
+        },
```
Comment (dtfiedler) [inline src/components/inputs/text/ArweavetransactionIdInput/ArweavetransactionIdInput.tsx:54]: i'm not following the need for this object it. we could simplify by having just a `Set` (or array) of validation errors

File: src/components/inputs/text/ValidationInput/ValidationInput.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  showValidationChecklist?: boolean;
+  placeholder?: string;
+
+  inputClassName?: string;
+  inputCustomStyle?: any;
+  disabled?: boolean; // disables input
+  value: string;
+  setValue: (text: string) => void;
+  setIsValid: (validity: boolean) => void;
+  setErrors?: (errors: Error[]) => void;
```
Comment (dtfiedler) [inline src/components/inputs/text/ValidationInput/ValidationInput.tsx:40]: what about `validationPredicate` as a prop, and that is a function that is passed in but returns errors if validation critiera is not passed.

**Comment by dtfiedler** [comment]
before digging into the implementation, running locally and seeing a few things
image

removed the text and the invalidations still show 
image

a valid ANT contract 
image

File: src/components/inputs/text/ValidationInput/ValidationInput.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}) {
+  const [validationResults, setValdidationResults] =
+    useState<ValidationObject[]>();
+
+  async function validationExecutor(id: string) {
+    setValue(id);
+    const validationResult: Promise<ValidationObject>[] = [];
+    validationPredicate.forEach((predicate) => {
+      validationResult.push(predicate(id));
+    });
```
Comment (dtfiedler) [inline src/components/inputs/text/ValidationInput/ValidationInput.tsx:42]:
```suggestion
    const validations: Promise<ValidationObject>[] = validationPredicate.map((predicate) => predicate(id));
```

File: src/components/inputs/text/ValidationInput/ValidationInput.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    const results = await Promise.all(validationResult);
```
Comment (dtfiedler) [inline src/components/inputs/text/ValidationInput/ValidationInput.tsx:43]: we'll want to use `Promise.allSettled` so it doesn't reject on the first one https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled

File: src/components/cards/ValidationList/ValidationList.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                      fill={'var(--text-faded)'}
+                    />
+                    Invalid Status Code
+                    <AlertCircle
+                      width={'16px'}
+                      height={'16px'}
+                      fill={'var(--text-white)'}
+                    />
+                  </span>
+                </Tooltip>
```
Comment (dtfiedler) [inline src/components/cards/ValidationList/ValidationList.tsx:83]: do we need the tool tip here? any input from lucas on this design wise?

File: src/components/cards/ValidationList/ValidationList.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                      width={20}
+                      height={20}
+                      fill={'var(--text-faded)'}
+                    />
+                    Invalid Status Code
+                    <AlertCircle
+                      width={'16px'}
+                      height={'16px'}
+                      fill={'var(--text-white)'}
+                    />
```
Comment (dtfiedler) [inline src/components/cards/ValidationList/ValidationList.tsx:81]: do we need the switch statment? it seems the only difference are the text and fill, can we do a ternary operator on those so this is less redundant?

File: src/components/cards/ValidationList/ValidationList.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    useState<Array<JSX.Element | undefined>>();
+
+  useEffect(() => {
+    if (validations) {
+      updateValidationList(validations);
+    }
+  }, [validations]);
+
+  function updateValidationList(validations: ValidationObject[]) {
+    const newElements = validations.map(
```
Comment (dtfiedler) [inline src/components/cards/ValidationList/ValidationList.tsx:18]: ths could also just go in the jsx below

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 8]
```diff
@@ -108,9 +111,10 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
     requiredTags?: { [x: string]: string[] };
   }): Promise<void> {
     // validate tx exists, their may be better ways to do this
+    // todo: implement http code error proccesor/handler
     const { status } = await this.getTransactionHeaders(id);
-    if (status !== 200) {
-      throw Error('Contract ID not found. Try again.');
+    if (!status.ok) {
+      throw Error(`Contract ID not found. Try again. Status: ${status}`);
```
Comment (dtfiedler) [inline src/services/arweave/SimpleArweaveDataProvider.ts:117]: this is showing errors even on 200's....
image

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 9]
```diff
@@ -141,4 +145,38 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
       });
     }
   }
+  async validateArweaveId(id: string): Promise<ValidationObject> {
+    const validatedIdObject = { name: '', status: false, error: '' };
```
Comment (dtfiedler) [inline src/services/arweave/SimpleArweaveDataProvider.ts:149]: i didn't intend for this type to be used here, only in the validtionInput component. 

these should throw an error, or return the transaction id

File: src/components/inputs/text/ValidationInput/ValidationInput.tsx (hunk 3 above)
Comment (atticusofsparta) [reply src/components/inputs/text/ValidationInput/ValidationInput.tsx:42]: facing some type errors on this, tbd in huddle....

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 10]
```diff
@@ -141,4 +145,38 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
...
+        requiredTags: {
+          'Contract-Src': approvedANTSourceCodeTxs,
+        },
+      });
+      validatedIdObject.status = true;
+      return validatedIdObject;
+    } catch (error: any) {
+      validatedIdObject.status = false;
+      validatedIdObject.error = error.message ? error.message : error;
+      return validatedIdObject;
```
Comment (dtfiedler) [inline src/services/arweave/SimpleArweaveDataProvider.ts:179]: throw the error, catch it and set the object error in the component

File: src/components/cards/ValidationList/ValidationList.tsx (hunk 5 above)
Comment (atticusofsparta) [reply src/components/cards/ValidationList/ValidationList.tsx:83]: Still discussing this with lucas but I mentioned and he seemed cool with it - dont have a solid answer yet.

File: src/types.ts [hunk 11]
```diff
@@ -91,6 +91,11 @@ export interface ArweaveDataProvider {
       [x: string]: string[] | ArweaveTransactionID[]; // allowed values
     };
   }): Promise<void>;
+  validateArweaveId(id: string): Promise<ValidationObject>;
+  validateAntContractId(
+    id: string,
+    approvedANTSourceCodeTxs: string[],
+  ): Promise<ValidationObject>;
```
Comment (dtfiedler) [inline src/types.ts:98]:
```suggestion
  validateArweaveId(id: string): Promise<ArweaveTransactionId>;
  validateAntContractId(
    id: string,
    approvedANTSourceCodeTxs: string[],
  ): Promise<ArweaveTransactionID>;
```

File: src/components/cards/ValidationList/ValidationList.tsx (hunk 6 above)
Comment (atticusofsparta) [reply src/components/cards/ValidationList/ValidationList.tsx:81]: Right now we dont, but, if we want to upgrade to a more advanced validation display in the future (like initial implementation with loading states) then I think a switch statement is the way to go.

**Review by dtfiedler — CHANGES_REQUESTED** [review]
a lot of cleanup - more than we need

File: src/components/cards/ValidationList/ValidationList.tsx (hunk 5 above)
Comment (dtfiedler) [reply src/components/cards/ValidationList/ValidationList.tsx:83]: it appears noisy and the check itself indicates if it passed/failed so feel it's redundant

File: src/components/cards/ValidationList/ValidationList.tsx (hunk 6 above)
Comment (dtfiedler) [reply src/components/cards/ValidationList/ValidationList.tsx:81]: maybe, but it's beefy and bloated for now. i'd prefer we create the proper object and then do the component rendering in JSX

File: src/components/layout/RegisterNameForm/RegisterNameForm.tsx [hunk 12]
```diff
@@ -157,14 +157,17 @@ function RegisterNameForm() {
                 }
                 placeholder={'Enter an ANT Contract ID Validation Input'}
                 showValidationChecklist={true}
-                validationPredicate={[
-                  (id: string) => arweaveDataProvider.validateArweaveId(id),
-                  (id: string) =>
+                validationPredicates={{
+                  'Is valid Arweave Transaction (TX) ID': (id: string) =>
```
Comment (dtfiedler) [inline src/components/layout/RegisterNameForm/RegisterNameForm.tsx:161]: nit: put these strings into an ENUM

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 13]
```diff
@@ -145,38 +145,33 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
...
+  async validateArweaveId(id: string): Promise<ArweaveTransactionID> {
+    return new Promise((resolve, reject) => {
+      try {
+        const txid = new ArweaveTransactionID(id);
+        resolve(txid);
+      } catch (error: any) {
+        reject(error);
+      }
+    });
+  }
```
Comment (dtfiedler) [inline src/services/arweave/SimpleArweaveDataProvider.ts:157]: nit: run lint/prettier to set proper spacing

File: src/components/cards/ValidationList/ValidationList.tsx [hunk 14]
```diff
@@ -103,7 +17,20 @@ function ValidationList({ validations }: { validations: ValidationObject[] }) {
           gap: '1em',
         }}
       >
-        {validationChecklist}
+        {validations?.map((validationItem: ValidationObject) => (
+          <span className="flex flex-row text flex-left white bold">
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/components/cards/ValidationList/ValidationList.tsx]: added

File: src/services/arweave/SimpleArweaveDataProvider.ts (hunk 13 above)
Comment (atticusofsparta) [reply src/services/arweave/SimpleArweaveDataProvider.ts:157]: ran

File: src/components/layout/RegisterNameForm/RegisterNameForm.tsx (hunk 12 above)
Comment (atticusofsparta) [reply src/components/layout/RegisterNameForm/RegisterNameForm.tsx]: added

File: src/services/arweave/SimpleArweaveDataProvider.ts [hunk 15]
```diff
@@ -145,38 +145,33 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
...
+  async validateConfirmations(id: string): Promise<void> {
+    const txid = new ArweaveTransactionID(id);
+    return this.validateTransactionTags({
```
[the remark this answers is not in the document]
Comment (atticusofsparta) [reply src/services/arweave/SimpleArweaveDataProvider.ts:160]: noted

File: src/hooks/useIsMobile/useIsMobile.tsx [hunk 16]
```diff
@@ -14,7 +14,8 @@ export default function useIsMobile(): boolean {
   }, [isMobile]);
 
   function updateIsMobile(): void {
-    setIsMobile(window.innerWidth < 600);
+    console.log('upating....', isMobile);
```
Comment (dtfiedler) [inline src/hooks/useIsMobile/useIsMobile.tsx:17]:
```suggestion
```

**Review by dtfiedler — APPROVED** [review]
LG - nice work!

**Pull request merged**